Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -255,87 +262,53 @@ void UPSWorldSubsystem::AddProgressionStarActors()
 void UPSWorldSubsystem::OnTakeActorsFromPoolCompleted(const TArray<FPoolObjectData>& CreatedObjects)
 {
 	const FPSRowData& CurrentRowData = GetCurrentRow();
+	float AmountUnlocked = CurrentRowData.CurrentLevelProgression;;
 
-	float CurrentAmountOfUnlocked = 0;
-	float CurrentAmountOfLocked = 0;
-
-	//set updated amount of stars
-	if (CurrentRowData.CurrentLevelProgression >= CurrentRowData.PointsToUnlock)
-	{
-		// set required points (stars)  to achieve for a level  
-		CurrentAmountOfUnlocked = CurrentRowData.PointsToUnlock;
-		CurrentAmountOfLocked = 0;
-	}
-	else
-	{
-		// Calculate the unlocked against locked points (stars) 
-		CurrentAmountOfUnlocked = CurrentRowData.CurrentLevelProgression;
-		CurrentAmountOfLocked = CurrentRowData.PointsToUnlock - CurrentRowData.CurrentLevelProgression;
-	}
-
-	float integerPart;
-	float fractionalPart;
-	// Setup spawned widget
 	for (const FPoolObjectData& CreatedObject : CreatedObjects)
 	{
-		if (CurrentAmountOfUnlocked > 0)
+		AActor& SpawnedActor = CreatedObject.GetChecked<AActor>();
+		UStaticMeshComponent* MeshComponent = SpawnedActor.FindComponentByClass<UStaticMeshComponent>();
+
+		SpawnedStarActorsInternal.Add(&SpawnedActor);
+
+		SpawnedActor.SetActorTransform(CurrentRowData.StarActorTransform);
+
+		// if the actor is the first element it set initial position
+		// from the initial position there is a distance between stars 
+		if (SpawnedStarActorsInternal.Num() > 1)
 		{
-			fractionalPart = modff(CurrentAmountOfUnlocked, &integerPart);
-			// check if it has a fractional part and if there is no more fully achieved stars 
-			if (fractionalPart < 1.0f && CurrentAmountOfUnlocked < 1.0f)
-			{
-				UpdateStarActor(CreatedObject, fractionalPart, 0);
-			}
-			else
-			{
-				UpdateStarActor(CreatedObject, 1, 0);
-			}
-
-			CurrentAmountOfUnlocked--;
-			continue;
+			SpawnedActor.SetActorLocation(SpawnedStarActorsInternal[SpawnedStarActorsInternal.Num() - 2]->GetActorLocation() + CurrentRowData.OffsetBetweenStarActors);
 		}
 
-		if (CurrentAmountOfLocked > 0)
+		float StarAmount = FMath::Clamp(AmountUnlocked, 0.0f, 1.0f);
+		if (AmountUnlocked > 0)
+		{
+			SetOrUpdateStarActorMesh(MeshComponent, StarAmount, false);
+		}
+		else
 		{
-			UpdateStarActor(CreatedObject, 0, 1);
-			CurrentAmountOfLocked--;
+			SetOrUpdateStarActorMesh(MeshComponent, 1, true);
 		}
+		AmountUnlocked -= StarAmount;
```

Comment:
[Request] Please, at the end of for-loop, give a call to the star actor like it's initialized. So you will have a place in future, where you can react on just spawned star, instead of continue growing this loop.
```
	for (const FPoolObjectData& CreatedObject : CreatedObjects)
	{
          APSStarActor& SpawnedActor = CreatedObject.GetChecked<APSStarActor>();
          //...
          // End of this loop
          SpawnedActor.OnInitialized();
        }
```
where `OnInitialized()` is your own function

Reply:
I was unable to remove completely logic with decreasing AmountUnlocked because I need to specify whether it is locked or unlocked stars in the calculation right now, achieved here https://github.com/h4rdmol/ProgressionSystem/pull/17/commits/1824aa449128a1e05fc57f22db63e1ea8dd2d5df

Reply:
@h4rdmol, OnInitialized does not need `Ref` argument, but `constref`. 
Therefore, last line `PreviousActorLocationRef = GetActorLocation();` also does nothing and might be removed as unused

### Expected result:
```cpp
void APSStarActor::OnInitialized(const FVector& PreviousActorLocation) // without ref in name, and `const`
```

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/17/commits/bd5b21b7f15fa01f174c59cb316fe15a6a756755